### PR TITLE
Feat/load programs

### PR DESCRIPTION
**Pull request by LucasUTNFRD**
# Load Programs Feature

Adds functionality to load LC-3 object files into VM memory.

## Changes
- Implement program loading from .obj files
- Add example programs in examples/
- Add tests for program loading
- Fix Clippy warnings and add documentation
- Unit tests verify correct program loading and memory addressing

### CODE REVIEW

**Review by tomip01 — APPROVED** [review]
LGTM

File: src/main.rs [hunk 1]
```diff
@@ -105,17 +144,115 @@ impl VM {
...
 
-    vm.run();
+    // TODO: Load the program into memory
+    match vm.load_program(filename) {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error loading program: {:?}", e);
+            std::process::exit(1);
+        }
+    }
```
Comment (LeanSerra) [inline src/main.rs]: if vm.load_program(filename).is_err() could be cleaner if you are not doing anything with the correct result

File: src/opdcodes.rs [hunk 2]
```diff
@@ -160,7 +162,7 @@ pub fn trap(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
             vm.state = VMState::Halted;
             Ok(())
         }
-        _ => std::process::exit(1),
```
Comment (LeanSerra) [inline src/opdcodes.rs:163]: I think you should return an error here.

File: src/main.rs [hunk 3]
```diff
@@ -105,17 +144,115 @@ impl VM {
 fn main() {
     // Read the program file given as the first command line argument
     // This will be used ./lc3-vm path/to/program.obj
-    // ensure that the argument is passed and the file is readable
     let args: Vec<String> = std::env::args().collect();
 
     if args.len() < 2 {
         eprintln!("Usage: {:?} [image-file1] ...", args.first());
```
Comment (LeanSerra) [inline src/main.rs]: Why the 3 dots in the usage string? Are there any other arguments we can set?

**Review by LeanSerra — CHANGES_REQUESTED** [review]
Looks good, I like the comments on every opcode, left some suggestions

File: src/main.rs (hunk 1 above)
Comment (LucasUTNFRD) [reply src/main.rs]: Done

File: src/opdcodes.rs (hunk 2 above)
Comment (LucasUTNFRD) [reply src/opdcodes.rs:163]: Done

File: src/main.rs (hunk 3 above)
Comment (LucasUTNFRD) [reply src/main.rs]: done

**Review by LeanSerra — APPROVED** [review]
LGTM

**Pull request merged**